packages/babel-helper-create-class-features-plugin/src/decorators.ts

```diff
@@ -181,7 +187,9 @@ function addProxyAccessorsFor(
   const { static: isStatic } = element.node;
 
   const thisArg =
-    version === "2023-05" && isStatic ? className : t.thisExpression();
+    (version === "2023-11" || version === "2023-05") && isStatic
```

Comment:
As a follow-up PR, we should update all these version checks to only happen in Babel 7.